modules/035-cni-flannel/images/flanneld/main.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+func main() {
+	iptablesMgr, err := iptables.NewWithProtocol(iptables.ProtocolIPv4)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = insertUnique(iptablesMgr, "filter", "INPUT", strings.Fields("-m conntrack --ctstate INVALID -j DROP"), 1)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if os.Getenv("POD_NETWORK_MODE") == "host-gw" {
+		err := deleteLinksByPrefix("flannel")
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
i think this if should be at the top of main func

Reply:
Done.